src/machine/machine_stm32_i2c.go

```diff
@@ -290,17 +262,17 @@ func (i2c I2C) controllerRequestWrite(addr addressable, option transferOption) e
 	return nil
 }
 
-func (i2c I2C) controllerReceive(addr addressable, r []byte) error {
-
-	if !i2c.waitForFlag(flagBUSY, false) {
-		return errI2CBusReadyTimeout
-	}
+func (i2c I2C) controllerReceive(addr uint16, r []byte) error {
 
 	// ensure peripheral is enabled
 	if !i2c.Bus.CR1.HasBits(stm32.I2C_CR1_PE) {
 		i2c.Bus.CR1.SetBits(stm32.I2C_CR1_PE)
 	}
 
+	if !i2c.waitForFlag(flagBUSY, false) {
+		return errI2CBusReadyTimeout
+	}
```

Comment:
Same here: this code has moved (perhaps unintentionally).

Reply:
These changes were intentional on my part to test if it fixed the problem being reported #1568 (which I can't reproduce), but it sounds like it had no effect. (Doesn't the original implementation order look backwards? It potentially checks for a peripheral flag before the peripheral is enabled....)

Reverted back to original ordering.